core/sail/fts/lucene-spin/src/test/java/org/eclipse/rdf4j/lucene/spin/LuceneSailSpinTest.java

```diff
@@ -48,89 +53,96 @@
  * @see <a href="https://github.com/eclipse/rdf4j/issues/235">issue #235</a>
+ * @see <a href="https://github.com/eclipse/rdf4j/issues/739">issue #739</a>
  */
-public abstract class AbstractLuceneSailSpinTest {
+public class LuceneSailSpinTest {
 
 	private static final String DATA = "org/eclipse/rdf4j/sail/220-example.ttl";
 
-	private static Logger log = LoggerFactory.getLogger(AbstractLuceneSailSpinTest.class);
+	private static Logger log = LoggerFactory.getLogger(LuceneSailSpinTest.class);
+
+	private File tempDir;
 
 	private Repository repository;
 
 	private RepositoryConnection connection;
 
-	private SearchIndex searchIndex;
-
 	@Before
 	public void setUp()
 		throws Exception
 	{
+
+		//activate sail debug mode
+		// System.setProperty("org.eclipse.rdf4j.repository.debug", "true");
 		// load data into memory store
 		MemoryStore store = new MemoryStore();
 
 		// add Support for SPIN function
 		SpinSail spin = new SpinSail(store);
-		spin.setEvaluationMode(TupleFunctionEvaluationMode.TRIPLE_SOURCE);
-
-		// add Lucene support
-		Properties parameters = new Properties();
-		configure(parameters);
-		searchIndex = LuceneSail.createSearchIndex(parameters);
-		spin.addQueryContextInitializer(new SearchIndexQueryContextInitializer(searchIndex));
-		repository = new SailRepository(spin);
+
+		// add Lucene Spin Sail support
+		LuceneSpinSail luc = new LuceneSpinSail(spin);
+		tempDir = Files.createTempDir();
+		log.debug("data file: {}", tempDir.getAbsolutePath());
+		luc.setDataDir(tempDir);
+		repository = new SailRepository(luc);
+
+		// set up parameters
+		configure(luc.getParameters());
+
 		repository.initialize();
+		// local connection used only for population
+		try (RepositoryConnection localConn = repository.getConnection()) {
+			localConn.begin();
+			populate(localConn);
+			localConn.commit();
+		}
 
-		connection = repository.getConnection();
-		populate(connection, searchIndex);
+		// local connection for verification only
+		try (RepositoryConnection localConn = repository.getConnection()) {
+			// validate population
+			//localConn.begin();
+			int count = countStatements(localConn);
+			log.trace("storage contains {} triples", count);
+			assert count > 0;
+			//localConn.commit();
+			localConn.close();
+		}
 
-		// validate population
-		int count = countStatements(connection);
-		log.info("storage contains {} triples", count);
-		assert count > 0;
+		// testing connection
+		connection = repository.getConnection();
+		connection.begin();
+		assert connection.isActive() : "connection is not active";
 	}
 
 	@After
 	public void tearDown()
-		throws IOException, RepositoryException
+		throws RepositoryException, IOException
 	{
-		try {
-			if (connection != null) {
-				connection.close();
-			}
-		}
-		finally {
-			try {
-				if (repository != null) {
-					repository.shutDown();
-				}
-			}
-			finally {
-				if (searchIndex != null) {
-					searchIndex.shutDown();
-				}
+		if (connection != null) {
```

Comment:
The try finally blocks above should not have been replaced with this unsafe version. The reasoning behind them is to isolate cross-test failures to more efficiently identify failing tests.